context.go

```diff
@@ -339,9 +344,8 @@ func (c *Context) Sqrt(d, x *Decimal) (Condition, error) {
 		// approx = 0.5 * (approx + f / approx)
 		ed.Mul(approx, tmp, decimalHalf)
 	}
-	p = c.Precision
-	nc.Precision = p
-	dp := int32(p)
+	nc.Precision = c.Precision
+	dp := int32(c.Precision)
 	approxsubhalf := new(Decimal)
```

Comment:
It's the same name as used in the paper. I think it's best to keep it as is because of that.